### PR TITLE
Change lead time representation in subs maproom

### PR DESCRIPTION
**Pull request by drewmresnick**
Based off of the email thread regarding lead times, I gathered that it did not matter directly how we represented the leads at least within the data itself, so my approach was:

Use the lead time representation and map those to some arbitrary lead time values that the user can configure both of in the configuration file. I tried to make it as flexible as possible to apply to other datasets.

Then in the maproom itself it allows the user to select from target date ranges.

### CODE REVIEW

File: subseas_flex_fcst/config-sample.yaml
Comment (drewmresnick) [inline subseas_flex_fcst/config-sample.yaml:19]: Would appreciate some help on better naming conventions here; my idea was that `dataLead` represents the data provider's lead representation, while `lead` represents how we use lead time to calculate the target dates

File: subseas_flex_fcst/layout.py
Comment (drewmresnick) [inline subseas_flex_fcst/layout.py:283]: Also this might have a better description than I wrote..

File: subseas_flex_fcst/maproom.py
Comment (drewmresnick) [inline subseas_flex_fcst/maproom.py:105]: Here also the dictionary index names could be improved; currently `provider` is the provider's lead time rep, while `numVal` is the numerical value we use for calculations

File: subseas_flex_fcst/layout.py
Comment (remicousin) [reply subseas_flex_fcst/layout.py:283]: Maybe: "Time period being forecasted"
Maproom text can elaborate on the relationship between S/L/target if need be.

File: subseas_flex_fcst/layout.py
Comment (remicousin) [inline subseas_flex_fcst/layout.py:256]: increase the Start Date box width so the date can fit in full.

File: subseas_flex_fcst/layout.py
Comment (remicousin) [inline subseas_flex_fcst/layout.py:289]: Increase width so that target date fits in full. Also the target date needs to contain the year information.

File: subseas_flex_fcst/config-sample.yaml
Comment (remicousin) [inline subseas_flex_fcst/config-sample.yaml:29]: could this be a dictionary in the first place? or whatever nested structured are called in config files? And that might save you from needing a numberLeads that will be the size of that list.

**Review by remicousin — COMMENTED** [review]
A superficial review for now. I'll have to look at it in more details later.

File: subseas_flex_fcst/config-sample.yaml
Comment (aaron-kaplan) [reply subseas_flex_fcst/config-sample.yaml:29]:
```
leads:
    wk1: 1
    wk2: 8
    wk3: 15
    wk4: 22
```

File: subseas_flex_fcst/layout.py
Comment (drewmresnick) [reply subseas_flex_fcst/layout.py:256]: For these issues I was trying to figure out a way to make the size of the text or the size of the div change to accommodate changing the window size, so that they would always show the  full dates. Not so much luck on that though.

Comment (remicousin) [reply subseas_flex_fcst/layout.py:256]: That would be nice but maybe not worth sweating too much over it. What could be nice is to have some clean code to print such dates range in a more concise manner when can be. e.g.
8 Apr 2022 - 14 Apr 2022
could be
8-14 Apr 2022.
Whereas 31 Dec 2022 - 6 Jan 2023 can only keep that format...
That would mean checking if the years are equal, then if the month are equals, and append the date string accordingly.

As for saving space, I can think of 2 ways:
1/ One could be to apply slightly smaller font size.
2/ Another could be to create some menu widget like uicore does, where the title of the control is put on top of it
http://iridl.ldeo.columbia.edu/maproom/Global/ForecastsS2S/precip_median_subx.html?S=0000%2023%20Sep%202022
Or a combination of both.

I would not make those a priority unless it appears to be trivial.

File: subseas_flex_fcst/maproom.py
Comment (remicousin) [reply subseas_flex_fcst/maproom.py:105]: Aaron provided an example on how to write the config file. This should bear on this code so I'll review after you had a chance to revisit it.

File: subseas_flex_fcst/config-sample.yaml
Comment (remicousin) [inline subseas_flex_fcst/config-sample.yaml:22]: probably safe to assume that taking the first lead by default will be ok in all circumstances. In which case can get rid of that config entry.

File: subseas_flex_fcst/layout.py
Comment (drewmresnick) [reply subseas_flex_fcst/layout.py:255]: Okay I just pushed a commit that prints the date ranges more concisely. I am not seeing how putting the title of the control over the top will impact the display of the dates within the dropdown; if this is an aesthetic change you might like to have done still I can make it into an issue to address later.

Comment (drewmresnick) [reply subseas_flex_fcst/layout.py:255]: Okay should I make an issue for this then or is it not much worth thinking about in general?

Comment (remicousin) [reply subseas_flex_fcst/layout.py:255]: I think we can let go. I am sure there will be plenty of opportunities to remind us we can do much better aesthetically.... :)

Comment (drewmresnick) [reply subseas_flex_fcst/layout.py:255]: Okay. Then I think this PR is ready for approval unless any other things are found by you are Aaron to be changed.

**Comment by drewmresnick** [comment]
@aaron-kaplan @remicousin Can I get one of your approvals to merge this if it looks good?

**Comment by remicousin** [comment]
Will look at it now.

File: subseas_flex_fcst/maproom.py
Comment (remicousin) [inline subseas_flex_fcst/maproom.py:108]: I would find simpler to read this if the date for the target start and the date for the target end were computed first once and for all (for each loop iteration). Then you run the tests to decide the format of the target start date. Note that the format of the target end date is always the same so need to include it in the test.
That would all save some lines.

**Review by remicousin — APPROVED** [review]
I approve in spite of my last comment, so that you can merge now or after acting upon it as you wish.
But you might want to wait for Aaron's review: he's the Python expert.

File: subseas_flex_fcst/maproom.py
Comment (aaron-kaplan) [reply subseas_flex_fcst/maproom.py:108]: I agree. It took me several minutes to figure out what was intended here; I think it would have been easier if the calculation were separate from the formatting.

Incidentally, I also find the term "issue date" clearer than "start date" because it eliminates the ambiguity/overloading of "start" to mean either S (issue date) or S + L (start of target period).

Comment (remicousin) [reply subseas_flex_fcst/maproom.py:108]: if we are it... I didn't want to load the boat (and I might be contradicting myself from the past here) but let's use the terms already in use (they probably what they are now for a good reason)

http://iridl.ldeo.columbia.edu/maproom/Global/ForecastsS2S/precip_flex_subx_weekly.html?S=0000%2023%20Sep%202022

I mean for the controls:
"Forecast Issued"
"Target Period"

Comment (drewmresnick) [reply subseas_flex_fcst/maproom.py:108]: Okay. Should I be changing the variable names throughout the code to reflect this?

**Comment by remicousin** [comment]
nice

File: subseas_flex_fcst/maproom.py
Comment (drewmresnick) [reply subseas_flex_fcst/maproom.py:108]: @remicousin Let me know if you would like me to make a new issue for changing variable names to reflect this difference, or if you just wanted it applied to the controls names.

Comment (remicousin) [reply subseas_flex_fcst/maproom.py:108]: Hopefully soon, we will have to generalize a good deal of the things here to work with seasonal forecast. That will be an occasion to rationalize names, and possibly easier to do or think of as we will have a new case to run this all through.

Comment (drewmresnick) [reply subseas_flex_fcst/maproom.py:108]: Okay then I will leave that for now.

**Pull request merged**
